chain/network/src/test_utils.rs

```diff
@@ -269,6 +269,29 @@ impl MockPeerManagerAdapter {
     pub fn put_back_most_recent(&self, request: PeerManagerMessageRequest) {
         self.requests.write().unwrap().push_back(request);
     }
+    /// Calls the handler for each message, but removing only those for which the handler returns
+    /// None (or else the returned message gets requeued; the returned message should be the same
+    /// as the one given). Returns true if any message was removed.
+    pub fn handle_filtered(
+        &self,
+        mut f: impl FnMut(PeerManagerMessageRequest) -> Option<PeerManagerMessageRequest>,
+    ) -> bool {
+        // We get the count first and then pop one by one, that way we avoid
+        // grabbing the lock for the whole duration, which might lead to a
+        // deadlock if the processing of the request results in more network
+        // messages.
+        let num_requests = self.requests.read().unwrap().len();
+        let mut handled = false;
+        for _ in 0..num_requests {
+            let request = self.requests.write().unwrap().pop_front().unwrap();
```

Comment:
Is `pop_front().unwrap()` actually fine here? If someone else called `handle_filtered` against it, it is possible that there will be no requests in the queue. Looks scary...

Reply:
I mean, this mock object is supposed to be drained by a single thread - the testing thread. Other threads can only push messages to it. Having two threads calling handle_filtered can't possibly lead to a good test even if this implementation was correct.

Reply:
Oh that's a test environment, I've missed that. Anyway worth fixing